### PR TITLE
Turning `didUserWin()` more readable and faster.

### PR DESCRIPTION
**Pull request by LordLucass**
1. Creating a `mapResultToAnimal()` function to handle the transformation of the results for animals;
2. Creating functions `hasTwoAnimals()` and `hasSequence()` to handle repetitive calculations and make the code leaner;
3. Using `map()` instead of `loop()` to transform the results into animals;
4. Avoiding repetitive code by using auxiliary functions.

PS. This is only a suggestion for improvement and should be tested according to the needs of the rest of the code.

### CODE REVIEW

**Comment by ySnoopyDogy** [comment]
Hey hey, sorry for the delayed response. I was in a vacation. I am creating some tests to validate this. Creating tests was something i wanna do from a while. Now it's the best moment to start.

**Comment by LordLucass** [comment]
Indeed. Enforcing code validation and adaptation through testing is a crucial factor when it comes to production applications. Good luck, whatever you need, at your disposal.

**Comment by ySnoopyDogy** [comment]
While I was creating the test I found out a bug with the sequence logic in bicho. I think it may affect your changes.
https://github.com/MenheraBot/MenheraBot/blob/master/packages/events/src/modules/bicho/finishBets.ts#L86

**Comment by LordLucass** [comment]
Can you be a little more specific? I don't think I understand it very well. How is it affected?

**Comment by ySnoopyDogy** [comment]
The loops were inverted.

```diff
- const hasTwoAnimals = animals.every((a) => user.includes(a));
+ const hasTwoAnimals = user.every((a) => animals.includes(a));
```

**Comment by LordLucass** [comment]
Sorry, I made a mistake when I ended up merging the codes and accidentally the old code went into the commit that should have been the new code. Allow me to correct it.

File: packages/events/src/modules/bicho/finishBets.ts [hunk 1]
```diff
@@ -64,33 +76,15 @@ const didUserWin = (results: number[][], option: string, bet: BichoBetType): boo
...
-      const hasTwoAnimals = user.every((a) => animals.includes(a));
-      if (!hasTwoAnimals) return false;
-
-      const firstIndex = animals.indexOf(user[0]);
-      const secondIndex = animals.indexOf(user[1]);
-
-      return firstIndex < secondIndex;
-    }
+    case 'corner':
+      return hasTwoAnimals(animals, option.split(' | '));
```
Comment (ySnoopyDogy) [inline packages/events/src/modules/bicho/finishBets.ts:82]: The test suits failed for this.

Looking at the old code, the corner case uses the loop
```ts
animals.every((a) => userChoices.includes(a));
```
While the sequence loop (hasTwoAnimals) use 
```ts
user.every((a) => animals.includes(a));
```

The function `hasTwoAnimals` is the old sequence loop. The corner loop needs to be different.


The test case which return an error was the check `vaca | vaca | vaca | vaca | vaca` in an array of results `vaca | águia | veado | leão | galo`

**Review by ySnoopyDogy — CHANGES_REQUESTED** [review]

File: packages/events/src/modules/bicho/finishBets.ts (hunk 1 above)
Comment (LordLucass) [reply packages/events/src/modules/bicho/finishBets.ts:82]: Got it. In fact, the code has this inconsistency. I believe that because `hasTwoAnimals()` follows a pattern, this modification will have to be directly in `switch`. I will do it here.

File: packages/events/src/modules/bicho/finishBets.ts [hunk 2]
```diff
@@ -64,33 +76,15 @@ const didUserWin = (results: number[][], option: string, bet: BichoBetType): boo
...
-    }
-
-    case 'corner': {
-      const animals = results.map((a) => {
-        const ten = Number(`${a[2]}${a[3]}`);
-        return BICHO_ANIMALS[Math.floor(ten / 4)];
-      });
-      const userChoices = option.split(' | ');
+    case 'corner':
       return animals.every((a) => userChoices.includes(a));
```
Comment (ySnoopyDogy) [inline packages/events/src/modules/bicho/finishBets.ts:82]: `userChoices` doesn't exists here

**Review by ySnoopyDogy — CHANGES_REQUESTED** [review]

**Review by ySnoopyDogy — DISMISSED** [review]
All tests passed! Thanks for the PR! Is it ok for you if I commit to this PR running a lint in the code?

**Comment by LordLucass** [comment]
Yes, there is no problem in standardizing the code to your stylization level. By the way, I apologize for the delay in responding and the inconsistencies in the code. Furthermore, I am glad that it contributes positively to the development of your application. You're welcome. 👍

**Comment by ySnoopyDogy** [comment]
No need to apologize for anything! Thanks again for the help

**Review by ySnoopyDogy — APPROVED** [review]
